.github/workflows/weekly_test.yml

```diff
@@ -0,0 +1,116 @@
+name: Weekly CDK and Terraform Test
+
+on:
+  schedule:
+    - cron: '0 9 * * 1'  # Run every Monday at 9:00 UTC
+  push:
+    branches:
+      - fix/typos-and-missing-instructions  # Temporary branch to test workflow
+
+jobs:
+  test-cdk-and-terraform:
+    runs-on: ubuntu-latest
+    timeout-minutes: 20
+    steps:
+    # Checkout code
+    - name: Checkout code
+      uses: actions/checkout@v2
+
+    # Install prerequisites
+    - name: setup awscli
+      uses: hsupu/setup-awscli@v1
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v1
+    - name: Setup Terraform
+      uses: hashicorp/setup-terraform@v3
+    - name: Setup NVM
+      uses: irby/setup-node-nvm@master
+      with:
+        node-version: 18
+    - name: APT dependencies
+      run: |
+        sudo apt-get update && sudo apt-get install -y \
+          build-essential \
+          libbz2-dev \
+          libssl-dev \
+          libreadline-dev \
+          libffi-dev \
+          zlib1g-dev \
+          libsqlite3-dev \
+          liblzma-dev
+    - name: Setup Pyenv
+      uses: "gabrielfalcao/pyenv-action@v17"
+      with:
+        default: 3.11
+    - name: Install Terraform CDK
+      run: |
+        npm install --global cdktf-cli@^0.18.0 aws-cdk-local aws-cdk
+    
+    # Install Python dependencies
+    - name: Create virtualenv for Python
+      run: |
+        pyenv install 3.11 && \
+        pyenv local 3.11 && \
+          python3 -m venv .venv && \
+          . .venv/bin/activate && \
+          pip3 install -r devops-tooling/requirements.txt
+
+    # Start Localstack
+    - name: Start Localstack
+      run: |
+        . .venv/bin/activate
+        make setup-aws
+
+        # Unset stale org secret
+        printenv
+        unset LOCALSTACK_API_KEY
+
+        # Use repo secret and start Localstack in background
+        echo "Starting Localstack"
+        LOCALSTACK_AUTH_TOKEN="${{ secrets.LOCALSTACK_AUTH_TOKEN }}" DOCKER_COMPOSE_FLAGS="-d --timeout=30" make start-localstack
```

Comment:
This starts LS with the docker-compose that maps ports to the host. We don't want to map anything to the host in a CI pipeline. We want LS and a dev container on the same docker network, completely isolated. And, we have to pass in a unique pipeline run key to use in the compose name so concurrent runs work. Maybe we should do all that in the next PR and just get this one baselined first?

Reply:
This is a good observation. On it.